xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp

```diff
@@ -233,7 +233,7 @@ bool CActiveAEBufferPoolResample::Create(unsigned int totaltime, bool remap, boo
       if (m_processor->GetChannelLayout().Count() > 2)                    /* Disable upmix for CActiveAEResample if DSP layout > 2.0, becomes perfomed by DSP */
         upmix = false;
 
-      m_dspBuffer = new CActiveAEBufferPool(m_inputFormat);               /* Get dsp processing buffer class, based on dsp output format */
+      m_dspBuffer = new CActiveAEBufferPool(m_format);               /* Get dsp processing buffer class, based on dsp output format */
```

Comment:
> please submit the better fix

Sure. But at first I like to roll out testbuilds. In view the community can be very helpful to test my changes on all platforms. Then I will create a new PR with the latest changes. Afterwards I like to work towards AudioDSP V2.

Also some add-ons have some issues and I like to provide the needed changes in one step.

The preview without sqashed commits is available [here](https://github.com/AchimTuran/xbmc/pull/2).